homeassistant/components/caldav/calendar.py

```diff
@@ -203,6 +211,57 @@ async def async_get_events(
         """Get all events in a specific time frame."""
         return await self.coordinator.async_get_events(hass, start_date, end_date)
 
+    async def async_create_event(self, **kwargs: Any) -> None:
+        """Create a new event in the calendar."""
+        _LOGGER.debug("Event: %s", kwargs)
+
+        event = Event()
+        event.add("summary", kwargs.get("summary"))
+        event.add("dtstart", kwargs.get("dtstart"))
+        event.add("dtend", kwargs.get("dtend"))
+        event.add("dtstamp", datetime.now(kwargs.get("tzinfo")))
+        event.add("description", kwargs.get("description"))
+        event.add("uid", str(uuid.uuid4()))
+
+        _rrule = kwargs.get("rrule")
+        if _rrule is not None and _rrule != "":
+            # If rrule is set as "" or None, icalendar errors out.
+            event.add("rrule", _rrule)
+
+        cal = Calendar()
+        cal.add("prodid", "-//homeassistant.io//CALDAV//EN")
+        cal.add("version", "2.0")
+        cal.add_component(event)
+        ics_data = cal.to_ical().decode("utf-8")
+
+        _LOGGER.debug("ICS data %s", ics_data)
+
+        await self.hass.async_add_executor_job(
+            self.coordinator.calendar.add_event, ics_data
+        )
+
+    async def async_delete_event(
+        self,
+        uid: str,
+        recurrence_id: str | None = None,
+        recurrence_range: str | None = None,
+    ) -> None:
+        """Delete an event on the calendar."""
+        _LOGGER.debug("Delete event: %s", uid)
+
+        def _delete_event() -> None:
+            """Search for an event and delete it."""
+            event = self.coordinator.calendar.search(uid=uid, event=True, expand=False)
+            if not isinstance(event, list):
+                _LOGGER.error("Expected a list of events, got %s", type(event))
```

Comment:
In general: It is not super helpful to swallow error and logs rather than raising an exception. See https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

Reply:
Deleting event doesn't exist anymore.